weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -1978,6 +1911,126 @@ def completions_create(
             response=res.response, weave_call_id=start_call.id
         )
 
+    # -------------------------------------------------------------------
+    # Streaming variant
+    # -------------------------------------------------------------------
+
+    def _setup_completion_model_info(
+        self, req: tsi.CompletionsCreateReq
+    ) -> tuple[str, Optional[str], str, Optional[str], dict[str, str], Optional[str]]:
```

Comment:
Non-blocking nit: define this as a standalone function and pass `model_info` as an input.